File: gfapy/line/edge/gfa1/oriented_segments.py
import gfapy

class OrientedSegments:

  @property
  def oriented_from(self):
    """
    Returns
    -------
    gfapy.OrientedLine
      The oriented segment represented by the from_segment/from_orient fields.
    """
    return gfapy.OrientedLine(self.from_segment, self.from_orient)

  @property
  def oriented_to(self):
    """
    Returns
    -------
    gfapy.OrientedLine
      The oriented segment represented by the to_segment/to_orient fields.
    """
    return gfapy.OrientedLine(self.to_segment, self.to_orient)

File: lib/rgfa/line/edge/gfa1/oriented_segments.rb
RGFA::Line::Edge::GFA1 ||= Module.new

module RGFA::Line::Edge::GFA1::OrientedSegments

  # @return [RGFA::OrientedLine] the oriented segment represented by the
  #   from/from_orient fields
  def oriented_from
    OL[from, from_orient]
  end

  # @return [RGFA::OrientedLine] the oriented segment represented by the
  #   to/to_orient fields
  def oriented_to
    OL[to, to_orient]
  end

end
